src/seedu/addressbook/commands/TagCommand.java

```diff
@@ -0,0 +1,67 @@
+package seedu.addressbook.commands;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.tag.Tag;
+
+/**
+ * Finds and lists all persons in address book whose possess the exact tag presented in argument keywords.
+ * Keyword matching is must be letter for letter. Non-case sensitive.
+ */
+public class TagCommand extends Command {
+
+    public static final String COMMAND_WORD = "tag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose have at least one of the tags specified "
+            + "in the keywords typed and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " friends";
+
+    private final Set<String> keywords;
+
+    public TagCommand(Set<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    /**
+     * Returns a copy of keywords in this command.
+     */
+    public Set<String> getKeywords() {
+        return new HashSet<>(keywords);
```

Comment:
Note that if you might want to return it as an immutable set if its not meant to be modifiable.